mood-analysis.rb

```diff
@@ -28,5 +28,65 @@ def analyze_mood(words)
   "05/11 Yay, yay, yay! I'm having a awfuly great day."
 ]
 
-puts analyze_mood(text[0])
-puts analyze_mood(text[1])
+# puts analyze_mood(text[0])
+# puts analyze_mood(text[1])
+
+def string_analysis(array)
+  array.each do |day|
+    p array[0][0..4] + analyze_mood(day)
+  end
+end
+
+# string_analysis(text)
+# 3. Write a method called `happy_days` to determine how many logged entries it takes until there have been three :-) happy days.
+def happy_days(words)
+  count =[]
+  days = 0
+  words.each do |day|
+    moods = analyze_mood(day)[-3..-1]
+    count << moods.include?(":-)")
+      days += 1
+    if count.length == 3
+      puts "It takes #{days} entries for 3 happy days to occur"
+      break
+    end
+  end
+  puts "sorry, to many moods!"
+end
```

Comment:
This works, but I would have preferred to see only the smallest piece of information necessary returned from this method. In this case, just a fixnum representing the number of days it took to hit 3 happy days, not a string with that information embeded inside. A single fixnum would allow us the opportunity to use this method in different ways instead of being confined to this specific use case.